tests/neptune/internal/channels/test_channels_values_sender.py

```diff
@@ -212,18 +213,21 @@ def test_send_when_waiting_for_next_value_timed_out(self):
         numeric_values = [ChannelValue(x=i, y=i, ts=self._TS + i)
                           for i in range(0, 3)]
 
+        # and
+        semaphore = threading.Semaphore(0)
+        # pylint: disable=protected-access
+        self._EXPERIMENT._send_channels_values.side_effect = lambda _: semaphore.release()
+
         # and
         channels_values_sender = ChannelsValuesSender(experiment=self._EXPERIMENT)
 
         # when
         for channel_value in numeric_values:
             channels_values_sender.send(self._NUMERIC_CHANNEL.name, self._NUMERIC_CHANNEL.channelType, channel_value)
 
-        # and
-        time.sleep(self.__TIMEOUT * 2)
-
         # then
         # pylint: disable=protected-access
+        semaphore.acquire()
```

Comment:
nice solution, is there a way to make it so that after some time the test fails? otherwise "failure" will mean indefinite blocking